src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,21 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  console.log()
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
```

Comment:
Without `.map` I will have to repeat `.getAttribute('href')` and `href.slice`. `a.href` (or `{ href }` destructuring) alone will not suffice because for some reason it return the complete URL when href attribute returns only the hash part. We could parse that URL, but really what's wrong with the maps, is it that big of a performance hit or something?